### PR TITLE
mempool: Limit gossip connections to persistent and non-persistent peers (experimental)

### LINKED ISSUE #1058: Analyze, prototype and experiment easily applicable solutions to reduce bandwidth usage
## Problem

The bandwidth consumption in CometBFT is high. In particular, this comes from the fact that the reactor of the mempool is a basic flooding gossip protocol (à la Bitcoin). Because the network topology is dense, it is very likely that a node receives a transaction multiple times from different neighbors. Notice that such a redundancy is useful to prevent byzantine or selfish nodes to stop the dissemination of a transaction in the network.

Some evidence of this redundancy is seen in #613

## Target audience

Consensus developers

## Upside

Small improvements could cut bandwidth usage in CometBFT by (at least) 25%, based on other results.

## Downside

There are many possible improvements to test. To mitigate the risk of spending too much time on discussing improvements, we will divide the improvements into two batches, one with quick wins and one with ideas that require further though and refinement.

- [x] #1086

For each of the approaches that are chosen on the brainstorm we will 

- prototype the approach
- experiment with improvements using e2e and fast prototyping
- experiment with the approaches that turned out promising in the QA infra
- document the solutions as ADRs

The remaining ideas will become part of #11 and if there is room to try these and other ideas on that issue, we will iterate on brainstorming and prototyping.

Consider if #1048 is pre-requisite for this task of if backlog.

## Definition of done

- Provide in the form of an ADR a new gossip protocol, or an improvement to the current one, that reduces the bandwidth of mempool transactions, as demonstrated by experiments on a testnet that resembles as close as possible to a real-world mainnet.
- Implement, test, and merge into main the solution proposed in the ADR.
- Protocols or ideas discussed but not implemented or prototyped should also be documented.

### PR DESCRIPTION
**Pull request by lasarojc**
Closes #1058

Complements the work done in #1558: adds two independent configuration values for limiting outbound gossip connections, for persistent peers and non-persistent peers. Unconditional peers are not affected by this feature.

---

#### PR checklist

- [ ] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: mempool/metrics.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ type Metrics struct {
 	AlreadyReceivedTxs metrics.Counter
 
 	// Number of connections being actively used for gossiping transactions
+	// excluding peristent peers.
```
Comment (lasarojc) [inline mempool/metrics.go:45]:
```suggestion
	// excluding persistent peers.
```

**Review by melekes — APPROVED** [review]
👍

**Comment by andynog** [comment]
Moved to `In Progress` again as per conversation with @hvanz, still need some work

**Review by lasarojc — COMMENTED** [review]
I cannot approve, since I opened the PR, but LGTM

File: config/config.go [hunk 2]
```diff
@@ -876,15 +876,19 @@ type MempoolConfig struct {
...
-	// If set to 0, this feature is disabled, that is, the number of active connections is not
-	// bounded.
-	// If enabled, a value of 10 is recommended based on experimental performance results using the
-	// default P2P configuration.
-	ExperimentalMaxUsedOutboundPeers int `mapstructure:"experimental_max_used_outbound_peers"`
+	// Experimental parameters to limit gossiping txs to up to the specified number of peers.
+	// We use two independent upper values for persistent peers and for non-persistent peers.
+	// Unconditional peers are not affected by this feature.
+	// If we are connected to more than the specified number of persistent peers, only send txs to
+	// the first ExperimentalMaxGossipConnectionsToPersistentPeers of them. If one of those
```
Comment (ValarDragon) [inline config/config.go:883]: This seems like it should not be the first 10, but instead a random 10 peers, no?

**Pull request merged**